src/main/flight/servos.c

```diff
@@ -296,20 +296,54 @@ STATIC_UNIT_TESTED void forwardAuxChannelsToServos(uint8_t firstServoIndex)
     }
 }
 
+static void biquadFilterInitd1d2(biquadFilter_t *filter, float sample)
+{
+    filter->d2 = filter->b2 * sample - filter->a2 * sample;
+    filter->d1 = filter->b1 * sample - filter->a1 * sample + filter->d2;
+}
+
 static void filterServos(void)
 {
-    if (servoConfig()->servo_lowpass_freq) {
+    static uint16_t rudderFilterFreq;
+
+    if (servoConfig()->servo_lowpass_freq || servoConfig()->tri_servo_lpf_hz) {
         // Initialize servo lowpass filter (servos are calculated at looptime rate)
         if (!servoFilterIsSet) {
             for (int i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
-                biquadFilterInitLPF(&servoFitlerState[i], servoConfig()->servo_lowpass_freq, gyro.targetLooptime);
+                biquadFilterInitLPF(&servoFilterState[i], servoConfig()->servo_lowpass_freq, gyro.targetLooptime);
+                biquadFilterInitd1d2(&servoFilterState[i], (float)servo[i]);
             }
+            rudderFilterFreq = servoConfig()->servo_lowpass_freq;
             servoFilterIsSet = true;
         }
 
+        bool rudderFilterEnable = false;
+        if (servoConfig()->tri_servo_lpf_hz) {
+            if (mixerConfig()->mixerMode == MIXER_TRI || mixerConfig()->mixerMode == MIXER_CUSTOM_TRI) {
+                int8_t throttlePercent = calculateThrottlePercent();
+                if (!ARMING_FLAG(ARMED) || (throttlePercent >=0 && throttlePercent < 20)) rudderFilterEnable = true;
+            }
+        }
         for (int i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
-            // Apply servo lowpass filter and do sanity cheching
-            servo[i] = (int16_t) biquadFilterApply(&servoFitlerState[i], (float)servo[i]);
+            if (i == SERVO_RUDDER && rudderFilterEnable) {
```

Comment:
I'd say both. Putting the duplicate code in a 3 line function makes it more readable anyway.

It's fine to exclude this feature from Naze to get it to build. You only need to put the `#ifdef` in `servo.c` and `cli.c`. It's better not to pollute the headers with build flags, and the linker will take care of eliminating unused functions.

I suggest you call the build flag `USE_TRI_SERVO_FILTER`